src/clients/HubPoolClient.ts

```diff
@@ -452,21 +454,41 @@ export class HubPoolClient {
         );
       }
     ) as ExecutedRootBundle;
-    if (mostRecentExecutedRootBundleEvent) {
-      // Arguably we don't need to even check these array lengths since we should assume that any proposed root bundle
-      // meets this condition.
-      if (
-        mostRecentExecutedRootBundleEvent.l1Tokens.length !== mostRecentExecutedRootBundleEvent.runningBalances.length
-      ) {
-        throw new Error("runningBalances and L1 token of ExecutedRootBundle event are not same length");
+
+    if (executedRootBundle) {
+      // runningBalances length must be a 1x or 2x multiple of l1Tokens length.
+      const { l1Tokens } = executedRootBundle;
+      let { runningBalances } = executedRootBundle;
+      let incentiveBalances: BigNumber[] = [];
+      if (runningBalances.length === l1Tokens.length) {
+        // Pre-UBA model (no incentives exist).
+        incentiveBalances = Array.from(runningBalances, () => toBN(0));
+      } else if (runningBalances.length === 2 * l1Tokens.length) {
+        // UBA model: runningBalances array is a concatenation of runningBalance and incentiveBalance.
+
+        runningBalances = runningBalances.slice(0, l1Tokens.length);
+        incentiveBalances = runningBalances.slice(l1Tokens.length);
+        this.logger.debug({
+          at: "HubPoolClient#getRunningBalanceBeforeBlockForChain",
+          message: "",
```

Comment:
Will this mean that the message is empty in the error output?

Reply:
That's a mistake :) Thanks for spotting it.

Reply:
Fixed now.